s3.go

```diff
@@ -1,33 +1,78 @@
 package veneur
 
 import (
+	"bytes"
+	"compress/gzip"
+	"encoding/csv"
 	"errors"
 	"io"
 	"path"
 	"strconv"
 	"time"
 
+	"github.com/DataDog/datadog-go/statsd"
+	"github.com/Sirupsen/logrus"
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/s3"
 	"github.com/aws/aws-sdk-go/service/s3/s3iface"
 )
 
+var _ plugin = &S3Plugin{}
+
+type S3Plugin struct {
+	logger *logrus.Logger
+	statsd *statsd.Client
+	svc    s3iface.S3API
+}
+
+func (p *S3Plugin) Flush(metrics []DDMetric, hostname string) error {
+	const Delimiter = '\t'
+	const IncludeHeaders = false
+
+	start := time.Now()
+	csv, err := encodeDDMetricsCSV(metrics, Delimiter, IncludeHeaders)
+	if err != nil {
+		log.WithFields(logrus.Fields{
```

Comment:
why use the package global here if there's a logger object inside the S3Plugin? ditto for other log statements in this file